display_manager.py

```diff
@@ -154,62 +157,54 @@ def _render_map(cls):
-                    if cur_map.fov[pos]:
-                        if wall:
-                            cls.console.draw_char(
-                                x, y, None, fg=None, bg=Colors.WALL_VISIBLE
-                            )
-                        else:
-                            cls.console.draw_char(
-                                x, y, None, fg=None, bg=Colors.GROUND_VISIBLE
-                            )
-                        # Tiles in FOV will be remembered after they get out
-                        # of sight, out of mind :^)
-                        cur_map.explored[pos] = True
-
-                    # Position is not visible, but has been explored before
-                    elif cur_map.explored[pos]:
-                        if wall:
-                            cls.console.draw_char(
-                                x, y, None, fg=None, bg=Colors.WALL_DARK
-                            )
-                        else:
-                            cls.console.draw_char(
-                                x, y, None, fg=None, bg=Colors.GROUND_DARK
-                            )
+        # if cls.dungeon.fov_recomputed:
+            # cls.dungeon.fov_recomputed = False
+
+        for x in range(cur_map.width):
+            for y in range(cur_map.height):
+                pos = Vector(x, y)
+                wall = not cur_map.transparent[pos]
+
+                # If position is visible, draw a bright tile
+                if cur_map.fov[pos]:
+                    if wall:
+                        sprites.add(Sprite(cls.sprites['wall_light'], x*16, y*16))
+                    else:
+                        sprites.add(Sprite(cls.sprites['floor_light'], x*16, y*16))
+                    # Tiles in FOV will be remembered after they get out
+                    # of sight, out of mind :^)
+                    cur_map.explored[pos] = True
+
+                # Position is not visible, but has been explored before
+                elif cur_map.explored[pos]:
+                    if wall:
+                        sprites.add(Sprite(cls.sprites['wall_dark'], x*16, y*16))
+                    else:
+                        sprites.add(Sprite(cls.sprites['floor_dark'], x*16, y*16))
+        sprites.draw(cls.screen)
 
     def _render_entities(cls):
         """
         Render visible entities by render layer to the buffer console.
         """
+        sprites = pygame.sprite.Group()
+
         entities_sorted = sorted(cls.dungeon.current_level.entities,
                                  key=lambda x: x.render_priority.value)
+
         for entity in entities_sorted:
             # Draw visible entities
-            if cls.dungeon.current_level.fov[entity.pos]:
-                cls.console.draw_char(
-                    entity.pos.x, entity.pos.y, entity.char, entity.color,
-                    bg=None
-                )
+            if cls.dungeon.current_level.fov[entity.pos] and not isinstance(entity.sprite, str):
```

Comment:
I did this for Stairs but I think it's no longer needed as the Entity's init function will try to load a file with the filename provided by the `sprite` argument and if it fails it will just create a square with the entity's color.